### ISSUE TITLE
Test failures for Preconnect link types

### ISSUE THREAD

**Issue by gregjopa** [Munter/hyperlink#123]
Preconnect links cause test failures like so:

```html
<!-- browser hint to open a connection to fonts.gstatic.com -->
<link rel="preconnect" href="https://fonts.gstatic.com">
```

    not ok 54 should respond with HTTP status 200
    operator: error
    expected: "200 https://fonts.gstatic.com"
    actual:   "404 https://fonts.gstatic.com"
    at: https://www.cars.com (6:30) <link rel="preconnect" href="https://fonts.gstatic.com">
        https://www.cars.com/ (8:30) <link rel="preconnect" href="https://fonts.gstatic.com">

I think this library should handle preconnect links differently (or at least ignore them). Perhaps it should do a dns lookup for them. Let me know your thoughts. I'm happy to send a PR.

**Comment by papandreou** [comment]
@gregjopa, thanks for reporting! I very much agree that we need a separate testing strategy for preconnect where we just attempt to look up the hostname in dns and connect to it. I've implemented that here: https://github.com/Munter/hyperlink/pull/124

**Comment by Munter** [comment]
Released in v3.0.1

**Comment by gregjopa** [comment]
I tested out v3.0.1 and it works great! Thanks for adding support for `preconnect`
